### PR TITLE
fix(kit): temporarily inline `lodash.template` from `lodash-es`

### LINKED ISSUE #20891: `lodash.template` is flagged up in snyk audit
### Environment

any

### Reproduction

Run package scanner like snyk to detect vulnerabilities.

https://security.snyk.io/vuln/SNYK-JS-LODASHTEMPLATE-1088054

### Describe the bug

Details:
https://security.snyk.io/vuln/SNYK-JS-LODASHTEMPLATE-1088054

While Nuxt itself may not be directly affected, this can cause unnecessary issues with security package scanners, posing a challenge for internal dependency approvals.

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by wilsonpinto**
### 🔗 Linked issue

#20891 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [X] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolves #20891 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by danielroe** [comment]
`lodash.template` as used in Nuxt is not vulnerable as it compiles module-provided templates with developer-provided content.

Adding `lodash-es` would add 2Mb+ to our dependencies which is not something I would ideally like - maybe inlining the template function would be an option

But I think there is action to publish the package: https://github.com/lodash/lodash/issues/5536.

**Comment by wilsonpinto** [comment]
Hi @danielroe,

I'm keen on resolving this issue as soon as possible given its impact on approving dependencies. Despite reaching out to the maintainers via email and creating an issue paired with a Merge Request to address the problem, I have yet to receive a response (+1 week).

The proposed changes in this pull request - https://github.com/lodash/lodash/pull/5649 - would effectively address the issue but require the package to be published in order for us to bump `lodash.template`.

I would like to ask you for advise on alternative ways to fix this if the maintainers are unable to update the version in a timely manner, my possible approaches:

1. Fork the project to create a new npm package, which we could then replace in nuxt instead of official `lodash.template`.
2. Incorporate "lodash.template" and its dependencies directly into @nuxt/kit, considering these seem to be internally used only and the functions are marked as deprecated.
3. Consider the use of lodash-es (maybe not an option).
4. Replace with another alternative dependency.

What do you think? I would appreciate your thoughts on these.

Thanks

**Comment by danielroe** [comment]
You've put some fantastic work into this, so I feel bad suggesting this. But what do you think about instead letting `unbuild` do the tree-shaking/inlining for us? It would be a matter of setting `failOnError: false` and adding `lodash-es` as a dev-dependency and updating the imports to it.

**Comment by wilsonpinto** [comment]
@danielroe no problem at all! I truly appreciate your follow-up on this matter. 

Would you mind making the changes for me? I'm not familiar with unbuild

Feel free to close this PR

**Review by danielroe — APPROVED** [review]

File: packages/kit/src/internal/template.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { promises as fsp } from 'node:fs'
-import lodashTemplate from 'lodash.template'
+// TODO: swap out when https://github.com/lodash/lodash/pull/5649 is merged
+import lodashTemplate from 'lodash-es/template'
```
Comment (wilsonpinto) [inline packages/kit/src/internal/template.ts:3]:
```suggestion
import { template as lodashTemplate } from 'lodash-es'
```

**Pull request merged**